cmd/zoekt-sourcegraph-indexserver/queue.go

```diff
@@ -252,6 +280,30 @@ func (q *Queue) SetIndexed(opts IndexOptions, state indexState) {
 	item.indexState = state
 	if state != indexStateFail {
 		item.indexed = reflect.DeepEqual(opts, item.opts)
+		item.consecutiveFailures = 0
+		item.backoffUntil = time.Unix(0, 0)
+	} else {
+		backoffDuration := time.Duration(item.consecutiveFailures+1) * q.backoffDuration
+
+		if backoffDuration > q.maxBackoff {
+			backoffDuration = q.maxBackoff
+		} else {
+			item.consecutiveFailures++
```

Comment:
I don't see why this should be unbounded if it is used just for identifying what the new backoff duration is (given the linear backoff solution currently).

If we want to log below the total consecutive failures then we'd increment this member unbounded. Although this count might be tracked already somewhere and I missed it.